cli/cmdlineparser.cpp

```diff
@@ -680,32 +665,27 @@ bool CmdLineParser::parseFromArgs(int argc, const char* const argv[])
                     const auto& excludedPaths = mSettings->project.guiProject.excludedPaths;
                     std::copy(excludedPaths.cbegin(), excludedPaths.cend(), std::back_inserter(mIgnoredPaths));
 
-                    const std::string platform(mSettings->project.guiProject.platform);
-
-                    if (platform == "win32A")
-                        mSettings->platform(Settings::Win32A);
-                    else if (platform == "win32W")
-                        mSettings->platform(Settings::Win32W);
-                    else if (platform == "win64")
-                        mSettings->platform(Settings::Win64);
-                    else if (platform == "unix32")
-                        mSettings->platform(Settings::Unix32);
-                    else if (platform == "unix64")
-                        mSettings->platform(Settings::Unix64);
-                    else if (platform == "native")
-                        mSettings->platform(Settings::Native);
-                    else if (platform == "unspecified" || platform == "Unspecified" || platform.empty())
-                        ;
-                    else if (!mSettings->loadPlatformFile(projectFile.c_str(), platform, mSettings->verbose) && !mSettings->loadPlatformFile(argv[0], platform, mSettings->verbose)) {
-                        std::string message("unrecognized platform: \"");
-                        message += platform;
-                        message += "\".";
-                        printError(message);
-                        return false;
+                    std::string platform(mSettings->project.guiProject.platform);
+
+                    // keep existing platform from command-line intact
+                    if (!platform.empty()) {
+                        if (platform == "Unspecified") {
+                            printMessage("'Unspecified' is a deprecated platform type and will be removed in Cppcheck 2.14. Please use 'unspecified' instead.");
+                            platform = "unspecified";
+                        }
+
+                        std::string errstr;
+                        const std::vector<std::string> paths = {projectFile, argv[0]};
+                        if (!mSettings->platform(platform, errstr, paths)) {
+                            printError(errstr);
+                            return false;
+                        }
                     }
 
-                    if (!mSettings->project.guiProject.projectFile.empty())
+                    if (!mSettings->project.guiProject.projectFile.empty()) {
+                        projectFile = mSettings->project.guiProject.projectFile;
```

Comment:
is this assignment needed?

Reply:
Yes. The subproject needs to be propagated so you get the proper project in the error messages.